### PR TITLE
Include instance ids in CC trace

### PR DESCRIPTION
**Pull request by abstratt**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheDebugLogIntegrationTest.groovy [hunk 1]
```diff
@@ -25,6 +25,37 @@ import static org.gradle.internal.cc.impl.fingerprint.ProjectSpecificFingerprint
 
 class ConfigurationCacheDebugLogIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
 
+    def "logs oid and hashes"() {
+        given:
+        buildFile """
+        def scriptObject = [foo: "bar"]
+        task ok { Task it ->
+            def someObject = scriptObject
```
Comment (bamboo) [inline platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheDebugLogIntegrationTest.groovy:33]: 💅 I'd inline `scriptObject` here

File: platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheDebugLogIntegrationTest.groovy [hunk 2]
```diff
@@ -25,6 +25,37 @@ import static org.gradle.internal.cc.impl.fingerprint.ProjectSpecificFingerprint
 
 class ConfigurationCacheDebugLogIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
 
+    def "logs oid and hashes"() {
+        given:
+        buildFile """
+        def scriptObject = [foo: "bar"]
```
Comment (bamboo) [inline platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheDebugLogIntegrationTest.groovy:31]: 💅 Indent script one level further than the enclosing `buildFile` call

File: platforms/core-configuration/graph-serialization/src/main/kotlin/org/gradle/internal/serialize/graph/Codec.kt [hunk 3]
```diff
@@ -84,9 +84,9 @@ fun <I, R> CloseableWriteContext.writeWith(
 
 interface Tracer {
 
-    fun open(frame: String)
+    fun open(frame: String, instance: Any? = null)
```
Comment (bamboo) [inline platforms/core-configuration/graph-serialization/src/main/kotlin/org/gradle/internal/serialize/graph/Codec.kt:87]: 💅 The default argument values are unnecessary

**Review by bamboo — CHANGES_REQUESTED** [review]

**Review by bamboo — APPROVED** [review]

File: platforms/core-configuration/graph-serialization/src/main/kotlin/org/gradle/internal/serialize/graph/LoggingTracer.kt [hunk 4]
```diff
@@ -31,19 +31,23 @@ class LoggingTracer(
...
-        log(frame, 'C')
+    override fun close(frame: String, instance: Any?) {
+        log(frame, instance, 'C')
     }
 
     private
-    fun log(frame: String, openOrClose: Char) {
+    fun log(frame: String, instance: Any?, openOrClose: Char) {
+        val oidAndHash = instance
+            ?.let { System.identityHashCode(it).toString(16) to it.hashCode().toString(16) }
```
[the remark this answers is not in the document]
Comment (abstratt) [reply platforms/core-configuration/graph-serialization/src/main/kotlin/org/gradle/internal/serialize/graph/LoggingTracer.kt:45]: I wonder what they are waiting for...

**Comment by abstratt** [comment]
Thanks for the reviews!

**Pull request merged**